ext/com_dotnet/com_typeinfo.c

```diff
@@ -58,24 +58,26 @@ PHP_MSHUTDOWN_FUNCTION(com_typeinfo)
  * b) a CLSID, major, minor e.g. "{00000200-0000-0010-8000-00AA006D2EA4},2,0"
  * c) a Type Library name e.g. "Microsoft OLE DB ActiveX Data Objects 1.0 Library"
  */
-PHP_COM_DOTNET_API ITypeLib *php_com_load_typelib(char *search_string, int codepage)
+PHP_COM_DOTNET_API ITypeLib *php_com_load_typelib(const zend_string *search_string, int codepage)
 {
 	ITypeLib *TL = NULL;
 	char *strtok_buf, *major, *minor;
+	char *token = estrndup(ZSTR_VAL(search_string), ZSTR_LEN(search_string));
 	CLSID clsid;
 	OLECHAR *p;
 	HRESULT hr;
 
-	search_string = php_strtok_r(search_string, ",", &strtok_buf);
+	token = php_strtok_r(token, ",", &strtok_buf);
 
-	if (search_string == NULL) {
+	if (token == NULL) {
 		return NULL;
 	}
 
 	major = php_strtok_r(NULL, ",", &strtok_buf);
 	minor = php_strtok_r(NULL, ",", &strtok_buf);
+	efree(token);
```

Comment:
I think you changed the behavior here, because previously search_string would get null terminated at the first `,`, now you're working on the original instead.